### ISSUE TITLE
QnAMakerDialog not working properly

### ISSUE THREAD

**Issue by GermanZz** [microsoft/BotBuilder-Samples#3620]
## Sample information
1. Sample type: samples
2. Sample language: Python
3. Sample name: 49.qnamaker-all-features

## Describe the bug
In this sample, there is a QnAMakerDialog class used. As per [documentation](https://docs.microsoft.com/en-us/python/api/botbuilder-ai/botbuilder.ai.qna.dialogs.qnamakerdialog?view=botbuilder-py-latest) it's possible to change a number of answers returned, card title, card no match response and etc. However, when I'm changing them, nothing changes.

## To Reproduce
1. Download "49.qnamaker-all-features"
2. Go to "app.py"
3. In QnAMakerDialog add parameter top=10
4. Start the bot
5. Write something which should return several answers

## Expected behavior
When I'm writing something, it should return several cards for me, so that I can select which is needed - like suggested actions.

**Comment by GermanZz** [comment]
Any updates on this one? Additional information needed? 
Please let me know if something is needed

**Comment by axelsrz** [comment]
Hello @GermanZz I was able to investigate this issue and found some interesting highlights: 

- The `top` parameter seems to be there just to expose the QnA api, which allows you to specify the first 'n' most likely answers for the question sent (which is conceptually different from the number of answers a question might have)
- Internally, the `top` parameter is successfully sent in the QnA call, changing the number of answers returned from the query if there are 'n' or more possible candidates as answers. **However, this is practically irrelevant as the `QnAMakerDialog` class will always return just the top answer.** This behavior seems to be consistent across C#/JS SDKs so I'm going to investigate if it's by design.
- From your issue description, I believe the feature you're looking for might be 'Add follow-up promtp' which would be found while building the KB in the QnA portal: 
![image](https://user-images.githubusercontent.com/5943395/145917630-8d776a99-0c38-48b7-a4ac-ad8dd9cdbbb0.png)

To summarize, I will investigate if this Dialog is designed to provide more than one answer as this is not the current behavior (not only in Python but in any of the SDKs). If you want a possible workaround, please look at the third point.

**Comment by tdurnford** [comment]
@axelsrz Any updates on this issue?

**Comment by carlosscastro** [comment]
@axelsrz could you share an update with where we are on this one?

**Comment by axelsrz** [comment]
Closing this issue as this behavior is consistent with QnA api, will decide if change design in the next version.
